src/wallet/wallet.cpp

```diff
@@ -3138,6 +3143,27 @@ int64_t CWallet::GetOldestKeyPoolTime()
     return oldestKey;
 }
 
+void CWallet::MarkDestinationTransactionsDirty(const CTxDestination target_dst)
+{
+    LOCK(cs_wallet);
+
+    for (auto& entry : mapWallet) {
+        CWalletTx& wtx = entry.second;
+
```

Comment:
nit, it could skip this `wtx` if already dirty, but not sure that's easy to know.

Reply:
true, but since `MarkDirty` just resets several values I think it's not so easy.

Reply:
```
bool dirty = false;
for (int i = 0; i < AMOUNTTYPE_ENUM_ELEMENTS; ++i) {
    dirty |= wtx.m_amounts[i].m_cached.none();
|
if (dirty) continue;
```
should work.

`CachableAmount` has `std::bitset m_cached` which sets various bits when something is cached. If no bits are set (checked with `none()`), nothing is cached so it is dirty and can thus be skipped.

Reply:
@achow101 nice, and what about `fChangeCached`?

Maybe this can be done next to not scope creep, could also add a `bool CWalletTx::m_is_dirty` to avoid going thru m_amounts.

Reply:
`fChangeCached` is only for computing the change amounts and isn't used by `GetCachableAmount` which is what we care about for marking a tx as dirty. I agree that we should follow that same model and have a variable to indicate whether the tx is marked dirty.

Reply:
Actually shouldn't the check be `&= wtx.m_amounts[i].m_cached.none()`? If all amounts are empty then no need to extract+markdirty?

Reply:
> I agree that we should follow that same model and have a variable to indicate whether the tx is marked dirty.

@achow done in #17889.

Reply:
Implement it, as @promag noted the check actually needed to look that all the cached values are reset, so I did that and restructed it a bit as well.

Reply:
I think you could revert this latest change considering what's done in #17889.